app/controllers/api/predictions_controller.rb

```diff
@@ -26,12 +43,13 @@ def authenticate_by_api_token
       @user = User.find_by_api_token(params[:api_token])
 
       unless valid_params_and_user?
-        render json: invalid_message, status: :unauthorized
+        render json: invalid_api_message, status: :unauthorized
       end
     end
 
-    def valid_params_and_user?
-      params[:api_token] && @user
+    def authorized_to_view_prediction?
+      return true unless @prediction.private?
```

Comment:
Maybe the early return could have been clearer, but I believe the logic is correct. `return true unless @prediction.private?` is another way of saying "If the prediction is not private, then the user is authorized to view the prediction."